lib/salus/scanners/yarn_audit.rb

```diff
@@ -127,6 +133,91 @@ def handle_legacy_yarn_audit
       report_failure
     end
 
+    def auto_fix_vulns
+      @vulns_w_paths.each do |vuln|
+        path = vuln["Path"]
+        package = vuln["Package"]
+        # vuln["Path"] for some indirect dependency "foo"
+        # looks like "foo > bar > baz".
+        # vuln["Path"] for some direct dependency "foo"
+        # looks like "foo"
+        is_direct_dep = path == package
+
+        # TODO: Fix indirect dependencies as well
+        fix_direct_dependency(vuln) if is_direct_dep
+      end
+    end
+
+    def fix_direct_dependency(vuln)
+      package = vuln["Package"]
+      patched_version_range = vuln["Patched in"]
+
+      vulnerable_package_info = run_shell("yarn info #{package} --json").stdout
+      begin
+        vulnerable_package_info = JSON.parse(vulnerable_package_info)
+      rescue StandardError
+        report_error("yarn info #{package} --json did not return JSON")
+        return
+      end
+      list_of_versions = vulnerable_package_info["data"]["versions"]
```

Comment:
it would be safer to check if `vulnerable_package_info["data"]` is nil.
You can do something like below, not sure if my syntax is right
```ruby
vulnerable_package_info["data"].&["versions"]
```